test/distributed/checkpoint/fsdp/test_fsdp_dsd.py

```diff
@@ -0,0 +1,88 @@
+# Owner(s): ["oncall: distributed"]
+
+import copy
+
+import torch
+import torch.nn as nn
+from torch.distributed._composable.fsdp import fully_shard
+from torch.distributed.checkpoint.state_dict import (
+    get_model_state_dict,
+    StateDictOptions,
+)
+from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
+from torch.testing._internal.common_fsdp import FSDPTest, MLP
+from torch.testing._internal.common_utils import run_tests
+
+
+class TestFullyShardWithDistributedStateDict(FSDPTest):
+    @property
+    def world_size(self) -> int:
+        return min(4, torch.cuda.device_count())
+
+    def _get_base_model(self, mlp_dim: int = 2):
+        base_model = nn.Sequential(
+            MLP(mlp_dim),
+            nn.Sequential(MLP(mlp_dim), nn.Linear(mlp_dim, mlp_dim)),
+            MLP(mlp_dim),
+        )
+        return base_model
+
+    @skip_if_lt_x_gpu(2)
+    def test_1d_fsdp_get_model_state_dict(self):
+        self.run_subtests(
+            {"mlp_dim": [2, 3, 4, 5]},
+            self._test_1d_fsdp_get_model_state_dict,
+        )
+
+    def _test_1d_fsdp_get_model_state_dict(self, mlp_dim: int):
```

Comment:
Should we test against a full training step to ensure the model state_dict work even after backward?

Reply:
Sounds good. I think I can add this in test_state_dict.py instead.